server/conn.go

```diff
@@ -804,6 +807,12 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	cc.lastPacket = data
 	cmd := data[0]
 	data = data[1:]
+	if util.EnablePProfSQLCPU.Load() {
+		defer pprof.SetGoroutineLabels(ctx)
+		lastSQL := getLastStmtInConn{cc}.String()
+		ctx = pprof.WithLabels(ctx, pprof.Labels("sql", parser.Normalize(lastSQL)))
+		pprof.SetGoroutineLabels(ctx)
```

Comment:
Just a question: if this goroutine starts some new goroutines, should we call this function with the same ctx label in these new goroutines when they start?

Reply:
I do some Investigate, new goroutines will inherit labels set by parent goroutine https://github.com/golang/go/blob/master/src/runtime/proc.go#L3466 :open_mouth:  maybe it's ok for us

Reply:
SO COOL...!